data_steward/constants/validation/participants/validate.py

```diff
@@ -229,62 +107,145 @@
+    ),
+    split_address_into_parts AS (
+        SELECT id, part_address
+        FROM standardize_apartment_number, UNNEST(SPLIT(address, ' ')) as part_address
+    ),
+    unabbreviate AS (
+        SELECT 
+            id, COALESCE(expansion, part_address) as normalized_part_address,
+        FROM split_address_into_parts p
+        LEFT JOIN address_abbreviations aa
+        ON aa.abbreviation = p.part_address
+    )
+    SELECT 
+        ARRAY_TO_STRING(ARRAY_AGG(normalized_part_address), ' ') as street,
+    FROM unabbreviate
+    GROUP BY id
+""")
+
+NORMALIZED_CITY = JINJA_ENV.from_string("""
+    WITH address_abbreviations AS (
+        SELECT *
+        FROM UNNEST(ARRAY<STRUCT<abbreviation STRING, expansion STRING>>[
+            {{abbreviation_tuples}}
+        ])
+    ),
+    remove_commas_and_periods AS (
+        SELECT 'dummy_id' as id, REGEXP_REPLACE(coalesce({{city}}, ''), '[,.]', '') as address
+    ),
+    remove_redundant_whitespaces AS (
+        SELECT id, REGEXP_REPLACE(TRIM(address), ' +', ' ') as address
+        FROM remove_commas_and_periods
+    ),
+    lowercase AS (
+        SELECT id, LOWER(address) as address
+        FROM remove_redundant_whitespaces
+    ),
+    split_address_into_parts AS (
+        SELECT id, part_address
+        FROM lowercase, UNNEST(SPLIT(address, ' ')) as part_address
+    ),
+    unabbreviate AS (
+        SELECT 
+            id, COALESCE(expansion, part_address) as normalized_part_address,
+        FROM split_address_into_parts p
+        LEFT JOIN address_abbreviations aa
+        ON aa.abbreviation = p.part_address
+    )
+    SELECT 
+        ARRAY_TO_STRING(ARRAY_AGG(normalized_part_address), ' ') as city,
+    FROM unabbreviate
+    GROUP BY id
 """)
 
 CREATE_STATE_COMPARISON_FUNCTION = JINJA_ENV.from_string("""
 CREATE FUNCTION IF NOT EXISTS
   `{{project_id}}.{{drc_dataset_id}}.CompareState`(rdr_state string, ehr_state string)
   RETURNS string AS ((
-        WITH normalized_rdr_state AS (
-            SELECT 
-              CASE
-                WHEN REPLACE(LOWER(TRIM(rdr_state)), 'piistate_', '') IN ({{state_abbreviations}})
-                THEN REPLACE(LOWER(TRIM(rdr_state)), 'piistate_', '')
-                WHEN rdr_state IS NULL THEN NULL
-                ELSE ''
-              END AS rdr_state
-        )
-        , normalized_ehr_state AS (
-            SELECT
-              CASE
-                WHEN LOWER(TRIM(ehr_state)) IN ({{state_abbreviations}})
-                THEN LOWER(TRIM(ehr_state))
-                WHEN ehr_state IS NULL THEN NULL
-                ELSE ''
-              END AS ehr_state
-        )
+    WITH normalized_rdr_state AS (
+        SELECT 
+            CASE
+            WHEN REPLACE(LOWER(TRIM(rdr_state)), 'piistate_', '') IN ({{state_abbreviations}})
+            THEN REPLACE(LOWER(TRIM(rdr_state)), 'piistate_', '')
+            WHEN rdr_state IS NULL THEN NULL
+            ELSE ''
+            END AS rdr_state
```

Comment:
If state_abbreviations can be in a table, you can join it and use COALESCE to simplify this logic